### ISSUE TITLE
[BITV]: Skip links are not working anymore

### ISSUE THREAD

**Issue by JuliaKirschenheuter** [nextcloud/server#41696]
Try to navigate via skip links to "main content" or "navigation". Nothing happens.

![Peek 2023-11-23 15-44](https://github.com/nextcloud/server/assets/6078378/3595ce15-10b2-458f-a8f7-b57fe78c5877)

**Comment by emoral435** [comment]
Do we know what the purpose of these buttons are supposed to do? Do they re-route to the top of the page, AKA, the main part of the page?

**Comment by JuliaKirschenheuter** [comment]
@emoral435 thank you for the question. More theory behind it: https://webaim.org/techniques/skipnav/#maincontent

For us it is important to have a quick access to the left navigation and to the main content. If user select "main content" then a next navigation point (via tab) have to start from the first element on the main content (like a table in a files view). If user would select "navigation" then the focus have to the first element in a left side navigation. It is just an anchor and helps quick navigation possibilities through a page.

**Comment by max-nextcloud** [comment]
It seems to depend on the app. Still working for me on collectives app for example. Not in files.

**Comment by susnux** [comment]
Apps that do not use the default `app-content` or `app-navigation` IDs need to set the correct IDs in their PHP view controller. For reference see the dashboard app.

I created a general solution for Vue apps (as all of them are affected): https://github.com/nextcloud-libraries/nextcloud-vue/pull/4983

**Comment by susnux** [comment]
Only fixed for files app - we still need the library PR.

**Comment by szaimen** [comment]
Seems to work in the files and dashboard app but not in the photos and activity app

**Comment by szaimen** [comment]
Update in activity app in stable28 seems to be missing still
